pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -133,30 +188,51 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 				defaultCluster.cluster.AltStatName = util.BuildStatPrefix(cb.push.Mesh.OutboundClusterStatName, string(service.Hostname), "", port, service.Attributes)
 			}
 
-			subsetClusters := cb.applyDestinationRule(defaultCluster, DefaultClusterMode, service, port, networkView)
+			subsetClusters := cb.applyDestinationRule(defaultCluster, DefaultClusterMode, service, port, clusterKey.networkView, clusterKey.destinationRule)
 
-			clusters = cp.conditionallyAppend(clusters, nil, defaultCluster.build())
-			clusters = cp.conditionallyAppend(clusters, nil, subsetClusters...)
+			if patched := cp.apply(nil, defaultCluster.build()); patched != nil {
+				cc := &discovery.Resource{Name: patched.Name, Resource: util.MessageToAny(patched)}
+				resources = append(resources, cc)
+				if features.EnableCDSCaching {
+					cb.cache.Add(clusterKey, tokens[clusterKey.clusterName], cc)
```

Comment:
It needs to be until your PR merged, right?

I would like to keep the 2 PRs decoupled since they are not dependant on each other. That way either one can merge first

Reply:
make sense